### DISCUSSION TITLE
Ragflow, add metadata to the documents and chunks

### DISCUSSION THREAD

**Discussion by devMls** [infiniflow/ragflow#3668] [Ideas]
Maybe this feature is present but I don't find that.

I miss can specify descriptions arround a document, for example in excels, write something like stoks from 2022 or stoks from 2024

In adittion in my case, I wish tell, this documents include corrections for this other. In laws something publish a correction for previous documents.

Other data like publish date can be usefull.

If this metadata is present in the chunks, the knowege for the LLM will be more productive.

**Comment by nordy1145** [comment]
Would add it would be beneficial, especially during the retrieval process to supply the metadata. Big piece of metadata that would be helpful is the source location, such as URL, file share location, etc. Would allow additional verification for users consuming the data.

**Comment by devMls** [comment]
I want to complete the idea.

I think that the metadata of the documents and the metada of the knowledge database should be inserted in the prompt. In this way:

> The knowledge database ${DATABASE_NAME} that has a description ${DATABASE_DESCRIPTION} has this documents with information:
>
> Document  ${DOCUMENT_NAME} with description ${DOCUMENT_DESCRIPTION}. This document has following attributes 
> ${METADATA1_NAME} : ${METADATA1_VALUE} 
> ${METADATA2_NAME} : ${METADATA2_VALUE} 
> This document  ${DOCUMENT_NAME} has the following information:
> ${CHUNK 1}
> ${CHUNK 2}
> End of ${DOCUMENT_NAME} 
> 
> Document  ${DOCUMENT_NAME} with description ${DOCUMENT_DESCRIPTION}. This document has following attributes 
> ${METADATA1_NAME} : ${METADATA1_VALUE} 
> ${METADATA2_NAME} : ${METADATA2_VALUE} 
> This document  ${DOCUMENT_NAME} has the following information:
> ${CHUNK 3}
> ${CHUNK 4}
> End of ${DOCUMENT_NAME} 
> 
> End of  ${DATABASE_NAME}
> 
> This other database ${DATABASE_NAME} that has a description ${DATABASE_DESCRIPTION} has this documents with information:
> 
> Document  ${DOCUMENT_NAME} with description ${DOCUMENT_DESCRIPTION}. This document has following attributes 
> ${METADATA1_NAME} : ${METADATA1_VALUE} 
> ${METADATA2_NAME} : ${METADATA2_VALUE} 
> This document  ${DOCUMENT_NAME} has the following information:
> ${CHUNK 5}
> ${CHUNK 6}
> End of ${DOCUMENT_NAME} 
> 
> Document  ${DOCUMENT_NAME} with description ${DOCUMENT_DESCRIPTION}. This document has following attributes 
> ${METADATA1_NAME} : ${METADATA1_VALUE} 
> ${METADATA2_NAME} : ${METADATA2_VALUE} 
> This document  ${DOCUMENT_NAME} has the following information:
> ${CHUNK 7}
> ${CHUNK 8}
> End of ${DOCUMENT_NAME} 
> 
> End of  ${DATABASE_NAME}


following this structure the LLM may have better context arround the knowledge.

In fact it is possible add metadata to the knowledge database.
